resteasy-core/src/main/java/org/jboss/resteasy/core/providerfactory/ResteasyProviderFactoryImpl.java

```diff
@@ -1021,17 +1021,16 @@ public <T extends Throwable> ExceptionMapper<T> getExceptionMapperForClass(Class
     //   @Override
     public <T> AsyncResponseProvider<T> getAsyncResponseProvider(Class<T> type) {
         Class asyncType = type;
```

Comment:
This could be removed and the usage below could just be replaced by the `type` argument.